### PR TITLE
Merged CSSE with master

### PR DESCRIPTION
**Pull request by jez4**

### CODE REVIEW

File: epifor/data/batch.py [hunk 1]
```diff
@@ -267,21 +269,32 @@ def export_region_traces(self, er: ExportRegion, out_dir: Path):
...
-            ]
+    def export_region_estimates(self, er: ExportRegion, df):
+
+        columns_list = sorted(set([date.split('_')[1] for date in df.columns]))
+        days = {}
+
+        try:
+            row = df.loc[er.region.key]
+        except Exception as ex:
+            logging.warning(f"Region not in data : {str(ex)}")
```
Comment (hnykda) [inline epifor/data/batch.py]: I propose to just catch `KeyError`

```suggestion
        except KeyError as ex:
            logging.warning(f"Region not in data : {str(ex)}")
```

and also there's `logging.exception`, but that may be overkill here

File: epifor/data/csse.py [hunk 2]
```diff
@@ -23,27 +24,61 @@
...
+    def convert_date(string):
+        " Convert string from csv to pandas allowed format"
+        try:
+            return datetime.datetime.strptime(string, "%m/%d/%Y").strftime("%Y%m%d")
+        except ValueError:
+            return datetime.datetime.strptime(string, "%m/%d/%y").strftime("%Y%m%d")
+
+    @staticmethod
+    def nearest_date(items, pivot):
+        " Find the nearest date to start_date from config "
```
Comment (hnykda) [inline epifor/data/csse.py]: Docstrings  should use `"""`, I know it's wrong in the rest of the module but I would say it's still better to stick to a standard for a new code.
```suggestion
        """Find the nearest date to start_date from config"""
```

File: epifor/data/csse.py [hunk 3]
```diff
@@ -23,27 +24,61 @@
...
         dfs = []
-        for name in ["Confirmed", "Deaths", "Recovered"]:
-            df = pd.read_csv(pattern.format(name.lower()), header=0)
+        shortest_list_dcs = None
+        by_date = by_date.strftime("%Y%m%d")
+        for name in ["confirmed", "deaths", "recovered"]:
+            df = pd.read_csv(pattern.format(name), header=0)
+
             dcs = list(df.columns)[4:]
+            # Rename columns to format pandas would accept
```
Comment (hnykda) [inline epifor/data/csse.py:55]: btw. pandas accepts almost any format, there are things like `pd.to_datetime` and you could also pass formatter when reading CSV. But if this works, it's fine

File: epifor/data/csse.py [hunk 4]
```diff
@@ -23,27 +24,61 @@
 class CSSEData:
     def __init__(self):
         self.df = None
-
-    def load(self, pattern):
+        self.HIST_FILE_NAME = "csse_history_data.h5"
```
Comment (hnykda) [inline epifor/data/csse.py]: this could be therefore a class attribute directly, given it's a constant, no?

File: gleambatch.py [hunk 5]
```diff
@@ -43,7 +53,11 @@ def update_data(args):
         log.info(f"Skipping Foretold update")
 
     log.info(f"Fetching/updating CSSE John Hopkins data from github ...")
-    run_command(["./fetch_csse.sh"])
+    # run_command(["./fetch_csse.sh"])
+    for name in ["confirmed", "deaths", "recovered"]:
+        result = urlopen(CSSE_URL.format(name)).read()
+        with open(CSSE_TARGET, "wb") as f:
```
Comment (hnykda) [inline gleambatch.py]: Can we write CSVs in non-binary mode?

**Review by hnykda — APPROVED** [review]
Would wait for @gavento review, otherwise fine with me

File: epifor/data/batch.py (hunk 1 above)
Comment (jez4) [reply epifor/data/batch.py]: ok

File: gleambatch.py (hunk 5 above)
Comment (jez4) [reply gleambatch.py]: Ok

File: epifor/data/csse.py (hunk 4 above)
Comment (jez4) [reply epifor/data/csse.py]: ok

File: epifor/data/csse.py (hunk 2 above)
Comment (jez4) [reply epifor/data/csse.py]: Corrected docstrings in files I have edited.

**Comment by gavento** [comment]
Looks good to me, will test now in data regen.

**Pull request merged**